sycl/include/sycl/device.hpp

```diff
@@ -291,6 +321,23 @@ class __SYCL_EXPORT device : public detail::OwnerLessBase<device> {
   template <backend BackendName, class SyclObjectT>
   friend auto get_native(const SyclObjectT &Obj)
       -> backend_return_t<BackendName, SyclObjectT>;
+
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  template <typename Param>
+  typename detail::is_device_info_desc<Param>::return_type
+  get_info_impl() const;
+
+  template <typename Param>
+  typename detail::GetInfoReturnType<
+      typename detail::is_device_info_desc<Param>::return_type>::type
+  get_info_internal() const;
+
+  // proxy of get_info_impl() to handle C++11-ABI compatibility separately.
+  template <typename Param> detail::string get_device_info() const;
+
+  template <typename Param>
+  std::vector<detail::string> get_device_info_vector() const;
```

Comment:
I still don't understand why it has to be 4 different functions instead of one...

Reply:
Sure, I reduced it to one. Thanks.